source/cpp/particles.cpp

```diff
@@ -80,6 +80,51 @@ int particles::add_particle(int c)
   return _cell2part[c].size() - 1;
 }
 //-----------------------------------------------------------------------------
+void particles::AddParticles(
+    Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic, Eigen::Dynamic,
+                                  Eigen::RowMajor>>
+        p_array,const std::vector<unsigned int>& p_template) 
```

Comment:
`AddParticles()` should not take `p_template` as an input argument. Rather, it should comply to the particle template that was passed in the constructor (i.e. `_ptemplate`). Note that a complicating factor here is that the advection class can modify/extend the particle template (see also below).